src/main/java/org/jboss/xavier/integrations/SpringConfiguration.java

```diff
@@ -0,0 +1,31 @@
+package org.jboss.xavier.integrations;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration
+@Profile("!test")
```

Comment:
When `test` profile is activated then `SpringConfiguration.java` is valid and also `TestConfigurationS3.java`, in theory, both of them should be valid, but I'm not sure if it will create an error or not... What about using `@Profile("default")` instead of `@Profile("!test")` what do you think?

Reply:
SpringConfiguration and TestConfigurationS3 are creating the AWS_S3 bean.
The test one needs to inject the URL of the host.
SpringConfiguration will only be enbled if profile is different than TEST ( note the ! in the profile name ).